### PR TITLE
Fix doctrine type mapping and sql declaration

### PR DESCRIPTION
**Pull request by dedpikhto**

### CODE REVIEW

File: src/DBAL/EnumType.php [hunk 1]
```diff
@@ -81,12 +81,18 @@ public function convertToDatabaseValue($value, AbstractPlatform $platform)
     /** {@inheritdoc} */
     public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
     {
-        return $platform->getVarcharTypeDeclarationSQL($fieldDeclaration);
+        return $this->name;
```
Comment (scaytrase) [inline src/DBAL/EnumType.php]: This change will break everything if there is no type alias present in database for this type

**Review by scaytrase — CHANGES_REQUESTED** [review]

**Comment by dedpikhto** [comment]
Review, please

File: src/DependencyInjection/LamodaEnumExtension.php [hunk 2]
```diff
@@ -22,10 +23,16 @@ public function load(array $configs, ContainerBuilder $container): void
         $container->register(IdenticalNamingStrategy::class, IdenticalNamingStrategy::class);
         $container->register(LowercaseNamingStrategy::class, LowercaseNamingStrategy::class);
 
+        if (isset($configs['enum_name_type_mapping'])) {
+            $enumNameTypeMapping = (bool)$configs['enum_name_type_mapping'];
+        } else {
+            $enumNameTypeMapping = false;
```
Comment (scaytrase) [inline src/DependencyInjection/LamodaEnumExtension.php]: we can use `??` here

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -15,6 +15,9 @@ public function getConfigTreeBuilder(): TreeBuilder
 
         $root = $builder->root('lamoda_enum');
 
+        $root->children()->scalarNode('enum_name_type_mapping')
```
Comment (scaytrase) [inline src/DependencyInjection/Configuration.php]: `booleanNode`

**Review by scaytrase — CHANGES_REQUESTED** [review]
Some minor comments. All other looks good

File: src/DependencyInjection/LamodaEnumExtension.php (hunk 2 above)
Comment (dedpikhto) [reply src/DependencyInjection/LamodaEnumExtension.php]: Already removed at all

File: src/DependencyInjection/Configuration.php (hunk 3 above)
Comment (dedpikhto) [reply src/DependencyInjection/Configuration.php]: fixed

**Comment by scaytrase** [comment]
Thank you, @dedpikhto

**Pull request merged**